api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -349,7 +351,7 @@ public Builder withSpecId(int newSpecId) {
 
     private Types.NestedField findSourceColumn(String sourceName) {
       Types.NestedField sourceColumn = schema.findField(sourceName);
-      Preconditions.checkArgument(sourceColumn != null, "Cannot find source column: %s", sourceName);
+      requireNonNull(sourceColumn, "Cannot find source column: " + sourceName);
```

Comment:
This isn't correct. Although the check is that the result of `findField` is null, the actual problem is that `sourceName` does not identify a column. That's why this throws `IllegalArgumentException` and not `NullPointerException`. It would be _very_ weird to pass in a non-null String and get a `NullPointerException` thrown.